task36/task36/Gamble.cpp

```diff
@@ -0,0 +1,50 @@
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+#include <algorithm>
+#include "Gamble.h"
+
+Gamble::Gamble() {
+    InitializeCriticalSection(&critsect);
+}
+
+Gamble::~Gamble() {
+    DeleteCriticalSection(&critsect);
+}
+
+void Gamble::AddPrediction(std::string userName, int prediction) {
+    EnterCriticalSection(&critsect);
+	this->predictions.insert(std::pair<std::string, int> (userName, prediction));
```

Comment:
instead of this-> use m_predictions

u can use std::make_pair() instead of std::pair